clouddrift/binning.py

```diff
@@ -0,0 +1,177 @@
+"""Module for binning Lagrangian data."""
+
+from collections.abc import Iterable
+
+import numpy as np
+import xarray as xr
+
+DEFAULT_BINS_NUMBER = 10
+
+
+def binned_statistics(
+    coords: np.ndarray | list[np.ndarray],
+    data: np.ndarray | list[np.ndarray] | None = None,
+    bins: int | list = DEFAULT_BINS_NUMBER,
+    bins_range: list | None = None,
+    dim_names: list[str] | None = None,
+    output_names: list[str] | None = None,
+    zeros_to_nan: bool = False,
+) -> xr.Dataset:
+    """
+    Compute N-dimensional histogram binning and calculate variable means in each bin.
+
+    Parameters
+    ----------
+    coords : array-like or list of array-like
+        Arrays of the coordinates of the Lagrangian data to be binned. For 1D data, a single array can be provided.
+        For multiple dimensions, the first array represents each coordinates along the first dimension,
+        the second array represents each coordinates along the second dimension, and so on.
+    data : array-like or list of array-like
+        Data associated at the Lagrangian coordinates described in coords. Multiple variables can be provided as a list.
+    bins : int or lists, optional
+        Number of bins per dimension (int) or bin edges per dimension (list). Default is 10.
```

Comment:
“Number of bins or bin edges per dimension. It can be:
	•	An int: same number of bins for all dimensions (default: 10),
	•	A list of ints or arrays: one per dimension, specifying either bin count or bin edges,
	•	None: defaults to 10 bins per dimension.”